packages/react-server/core/renderMiddleware.js

```diff
@@ -894,8 +904,8 @@ function setupLateArrivals(res) {
 	});
 
 	// TODO: maximum-wait-time-exceeded-so-cancel-pending-requests code
-	var promises = notLoaded.map( result => result.entry.dfd.promise );
-	RLS().lateArrivals = Q.allSettled(promises)
+	var promises = notLoaded.map( result => result.entry.promise );
+	RLS().lateArrivals = Promise.all(promises).catch(() => {})
```

Comment:
Yeah, that's a good idea.

Now that I look at the documentation for [`Promise.all`](http://bluebirdjs.com/docs/api/promise.all.html) it looks like this actually isn't equivalent to `Q.allSettled()`, since `Promise.all` _short circuits_ on rejection, whereas `Q.allSettled()` waits for resolution _or rejection_ of all promises.

There were a few `Q.allSettled` calls that I ported this way.  I'll need to come up with a better equivalent.